sansa-owl-spark/src/main/scala/net/sansa_stack/owl/spark/stats/OWLStats.scala

```diff
@@ -0,0 +1,821 @@
+      .asInstanceOf[RDD[OWLInverseFunctionalObjectPropertyAxiom]]
+      .map(a => a.getProperty)
+      .filter(_.isNamed)
+      .map(_.getNamedProperty)
+
+    val reflexiveFuncObjProps: RDD[OWLEntity] = owlAxioms.extractAxioms(axioms, AxiomType.REFLEXIVE_OBJECT_PROPERTY)
+      .asInstanceOf[RDD[OWLReflexiveObjectPropertyAxiom]]
+      .map(a => a.getProperty)
+      .filter(_.isNamed)
+      .map(_.getNamedProperty)
+
+    val irreflexiveFuncObjProps: RDD[OWLEntity] = owlAxioms.extractAxioms(axioms, AxiomType.IRREFLEXIVE_OBJECT_PROPERTY)
+      .asInstanceOf[RDD[OWLIrreflexiveObjectPropertyAxiom]]
+      .map(a => a.getProperty)
+      .filter(_.isNamed)
+      .map(_.getNamedProperty)
+
+    val symmetricObjProps: RDD[OWLEntity] = owlAxioms.extractAxioms(axioms, AxiomType.SYMMETRIC_OBJECT_PROPERTY)
+      .asInstanceOf[RDD[OWLSymmetricObjectPropertyAxiom]]
+      .map(a => a.getProperty)
+      .filter(_.isNamed)
+      .map(_.getNamedProperty)
+
+    val asymmetricObjProps: RDD[OWLEntity] = owlAxioms.extractAxioms(axioms, AxiomType.ASYMMETRIC_OBJECT_PROPERTY)
+      .asInstanceOf[RDD[OWLAsymmetricObjectPropertyAxiom]]
+      .map(a => a.getProperty)
+      .filter(_.isNamed)
+      .map(_.getNamedProperty)
+
+    val transitiveObjProps: RDD[OWLEntity] = owlAxioms.extractAxioms(axioms, AxiomType.TRANSITIVE_OBJECT_PROPERTY)
+      .asInstanceOf[RDD[OWLTransitiveObjectPropertyAxiom]]
+      .map(a => a.getProperty)
+      .filter(_.isNamed)
+      .map(_.getNamedProperty)
+
+    val typedOWLEntities: RDD[OWLEntity] = spark.sparkContext
+      .union(declarations, classAssertion, funcDataProps, funcObjProps,
+        inverseFuncObjProps, reflexiveFuncObjProps, irreflexiveFuncObjProps,
+        symmetricObjProps, asymmetricObjProps, transitiveObjProps)
+      .distinct()
+
+    typedOWLEntities
+  }
+
+  /**
+    * Criterion 25. Labeled subjects criterion.
+    *
+    * @param axioms RDD of triples
+    * @return RDD of labeled OWL annotation subjects.
+    */
+  def getLabeledSubjects(axioms: RDD[OWLAxiom]): RDD[OWLAnnotationSubject] = {
+    val annPropAssertion = owlAxioms.extractAxioms(axioms, AxiomType.ANNOTATION_ASSERTION)
+                                    .asInstanceOf[RDD[OWLAnnotationAssertionAxiom]]
+
+    annPropAssertion
+      .filter(a => a.getProperty.isLabel)
+      .map(_.getSubject)
+  }
+
+  /**
+    *  Criterion 26. SameAs axioms
+    *
+    * @param axioms RDD of OWLAxioms
+    * @return The number of OWL same individuals axioms in the input RDD
+    */
+  def getSameAsAxiomsCount(axioms: RDD[OWLAxiom]): Long = owlAxioms.extractAxioms(axioms, AxiomType.SAME_INDIVIDUAL).count()
+
+  /**
+    * Criterion 27. Links.
+    *
+    * Computes the frequencies of links between entities of different namespaces.
+    * This measure is directed, i.e. a link from `ns1 -> ns2` is different from `ns2 -> ns1`.
+    *
+    * TODO: Handle cases of anonymous OWL objects and/or explain here which cases are omitted and why
```

Comment:
Throughout this method the assumption was made that all OWL objects (i.e. property expressions, class expressions, OWL individuals) are named/not anonymous. So namespace links between non-named OWL objects are not considered. This should be either explained and justified in the method's doc string or implemented.

Reply:
for both LODStats and DistLodStats they only consider the triples with URI and avoid the blank nodes in calculations. So I follow the same procedure and avoid the anonymous OWL objects. will mention that in the comment.